dotnet/src/SemanticKernel.Abstractions/Functions/KernelPluginMetadata.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+#pragma warning disable IDE0130
+// ReSharper disable once CheckNamespace - Using the main namespace
+namespace Microsoft.SemanticKernel;
+#pragma warning restore IDE0130
+
+/// <summary>
+/// Provides read-only metadata for an <see cref="IKernelPlugin"/>.
+/// </summary>
+public sealed class KernelPluginMetadata
```

Comment:
Should we consider using record instead?

```suggestion
public sealed record KernelPluginMetadata
```